### ISSUE TITLE
9th List of problems with screenshots

### ISSUE THREAD

**Issue by supercoolsonic** [PiKeyAr/sadx_dreamcast#74]
Had a bit of time last night to discover some problems.

1. I don't know if this effects anything else, but I found another Super Sonic aura layering problem.
![missing aura laying issue](https://cloud.githubusercontent.com/assets/26051785/23805818/0d484740-0585-11e7-8bff-14a78a3a50fe.png)
It's at the end of Final Egg Act 3.

2. The ice walls in Ice Cap Act 2 have a layering problem when viewed from certain angles.
![changing ice](https://cloud.githubusercontent.com/assets/26051785/23805851/28796a4e-0585-11e7-9a03-0939feae4aab.png)
I remember using SADXLVL to restore the ice walls in SADX a long time ago, and the same thing happened then, too. This also happens with the frozen pond at the bottom of the Act, as well.

3. The ripples the air bubbles make when they reach the surface of the water here in Emerald Coast Act 2 cause z-fighting and appear to have some alpha weirdness.
![z-fighting ripples of a bubble](https://cloud.githubusercontent.com/assets/26051785/23805909/61d1c71e-0585-11e7-82d7-c592e5d4c83a.png)

4.  There's a slight problem with the lights around the shower sign in Casinopolis.
![slight alpha problem](https://cloud.githubusercontent.com/assets/26051785/23805970/8b4c8bba-0585-11e7-89ff-a1de95127f06.png)

5. This one is a bit more difficult to explain. The texture in center of these squares don't seem to react to the "moving light effect" going around the room.
![squares are not being lit](https://cloud.githubusercontent.com/assets/26051785/23806014/aefbb4fa-0585-11e7-882c-072795af20b5.png)
To be specific, I'm referring to the texture inside the gold border. It remains dark.

This is not an issue on the Dreamcast version.
![squares are being lit](https://cloud.githubusercontent.com/assets/26051785/23806048/d0085388-0585-11e7-8f3a-15ebaaa04dc2.png)
It's very hard to notice, but I'm not sure if it's a problem with the lighting mod, or the DC mod.

6. The texture on the flags in Twinkle Park is a little goofed up.
SADX:
![bad flags](https://cloud.githubusercontent.com/assets/26051785/23806115/1256fa8c-0586-11e7-93f2-e1995c81b9fd.png)

Dreamcast:
![proper flags](https://cloud.githubusercontent.com/assets/26051785/23806124/1af4716a-0586-11e7-8406-89e54a507be8.png)

7. The lights on the underside of the top part of the merry-go-round are a little different between SADX and Dreamcast.
SADX:
![merry-go-round lights sadx](https://cloud.githubusercontent.com/assets/26051785/23806216/6a26f1b8-0586-11e7-9022-f2bd4649925d.png)

Dreamcast:
![merry-go-round lights](https://cloud.githubusercontent.com/assets/26051785/23806222/6f2e3658-0586-11e7-98ca-f686b288b5c6.png)

What's interesting though, is that it appears that the lights on SADX could just be flipped the other way, "connecting" the texture to form a full light. The Dreamcast version doesn't look correct, but just thought I'd throw this out there.

8. The color of the shadow underneath the info monitor at the start of Twinkle Park Act 1 is grey currently while it's black on Dreamcast.

As soon as I finish my college papers, I'll get right to work on carefully shifting through your SET files and fixing a few things. Have a great day!

**Comment by Zetchzie** [comment]
Looks like the horses have a different lighting going on in DC. Good luck on your work ^^

**Comment by Zetchzie** [comment]
Funny thing is 7 doesn't happen on the GC port

**Comment by supercoolsonic** [comment]
Confirmed. 7 does not occur on the Gamecube version.
![merry-go-round gamecube](https://cloud.githubusercontent.com/assets/26051785/23808044/cb3ab212-058d-11e7-8483-32ff73d7af93.png)

The entire merry-go-round object may as well be ported from the Dreamcast. There are texture differences on the horses, and the environment maps aren't there, and the lighting is different.

**Comment by Zetchzie** [comment]
Yea the flags are fine in the GC version aswell.

**Comment by supercoolsonic** [comment]
I believe the red carpet is part of the merry-go-round model itself, which would explain the differences. I just checked the banners on vanilla SADX on the PC, and it doesn't have the problem. It's a problem with changing the textures. Either the texture needs to be changed, the Dreamcast model needs to be ported, or there's something else I'm missing.

I also checked the merry-go-round, and the errors I talked about still happen on vanilla SADX on PC.

**Comment by Zetchzie** [comment]
Maybe something for the others to look at but it looks like if you could lower the spinning ground that would allow the full textures to show but that may be oversimplifying the issue on my end. I think the lights can be fixed on PkRs end he may know what's wrong

**Comment by supercoolsonic** [comment]
On my lunch break, I have some free time. And using that free time, I determined that I'm wrong. The red carpet is not actually part of the level model. Thanks for bringing this up. I didn't notice this. I compared the SET files side-by-side, and determined that the merry-go-round is in the exact same place in both versions. Therefore, the conclusion is that the carpet is separate from the object, and that the object model is different between SADX and Dreamcast.

**Comment by PiKeyAr** [comment]
1. I sorted the models in that area, not sure if that helps but that's as much as I'm willing to do for it.
2. Fixed. There's a few other glitches involving that object, but those aren't really fixable.
3. This is only fixable with a custom draw queue, which may or may not happen soon.
4. Fixed.
5. I have no idea why that would be happening since it's exactly the same material and it has no "ignore specular" or "ignore light" flags, and there's no alpha involved. You gotta keep in mind the light effect in Casino isn't a 1:1 port of the original code, it was imitated from scratch by the author of the PL mod.
6. Fixed.
7. Fixed. The models in this area use the same UVs as their Dreamcast counterparts, and I suspect the models themselves are pretty much identical. However, they mirrored that light texture and forgot to update this and a few other models. To fix this, I had to restore Dreamcast UVs on several other objects (bowling room door, pirate ships etc.). The spinning floor model is identical with the Dreamcast model, but it's set up differently in code, so I had to hide its model and implement my own. This also fixed shadow Z-fighting with the spinning spiked balls.
8. Not caused by this mod. See here: https://github.com/SonicFreak94/sadx-dc-lighting/issues/22
